File: typescript/src/n0232_implement_queue_using_stacks.ts
export class MyQueue {
  input: number[];
  output: number[];

  constructor() {
    this.input = [];
    this.output = [];
  }

  push(x: number): void {
    this.input.push(x);
  }

  pop(): number {
    if (this.output.length === 0) {
      while (this.input.length > 0) {
        this.output.push(this.input.pop()!);
      }
    }
    return this.output.pop()!;
  }

  peek(): number {
    if (this.output.length === 0) {
      while (this.input.length > 0) {
        this.output.push(this.input.pop()!);
      }
    }
    return this.output[this.output.length - 1]!;
  }

  empty(): boolean {
    return this.input.length === 0 && this.output.length === 0;
  }
}

File: python/src/n0232_implement_queue_using_stacks.py
class MyQueue:
    def __init__(self) -> None:
        self.input: list[int] = []
        self.output: list[int] = []

    def push(self, x: int) -> None:
        self.input.append(x)

    def pop(self) -> int:
        if not self.output:
            while self.input:
                self.output.append(self.input.pop())
        return self.output.pop()

    def peek(self) -> int:
        if not self.output:
            while self.input:
                self.output.append(self.input.pop())
        return self.output[-1]

    def empty(self) -> bool:
        return not self.input and not self.output
